app/components/issues.jsx

```diff
@@ -0,0 +1,174 @@
+    httpRequest.onreadystatechange = function() {
+      var data = {};
+      if (httpRequest.readyState === 4) {
+        if (httpRequest.status === 200) {
+          data = JSON.parse(httpRequest.responseText);
+          callback(data);
+        }
+      }
+    };
+    httpRequest.open('GET', url);
+    httpRequest.send();
+  },
+  getInitialState: function() {
+    return {
+      search: "assignee",
+      term: "",
+      query: [],
+      users: [],
+      repos: [],
+      milestones: [],
+      labels: []
+    };
+  },
+  onSearchChange: function(e) {
+    this.setState({
+      search: e.target.value
+    });
+  },
+  onSubmit: function(e) {
+    e.preventDefault();
+    var query = this.state.query;
+    query.push({
+      search: this.state.search,
+      searchTerm: this.state.search + ":" + this.state.term
+    });
+    this.setState({
+      query: query,
+      term: ""
+    });
+  },
+  onChange: function(event) {
+    this.setState({
+      term: event.target.value
+    });
+  },
+  componentDidMount: function() {
+    this.getJson("/api/github/foundation-users", function(data) {
+      this.setState({
+        users: data
+      });
+    }.bind(this));
+
+    this.getJson("/api/github/mozilla-repo-names", function(data) {
+      this.setState({
+        repos: data
+      });
+    }.bind(this));
+
+    this.getJson("/api/github/mozilla-labels", function(data) {
+      this.setState({
+        labels: data
+      });
+    }.bind(this));
+
+    this.getJson("/api/github/mozilla-milestones", function(data) {
+      this.setState({
+        milestones: data
+      });
+    }.bind(this));
+  },
+  render: function() {
+    var query = this.state.query;
+    var term = this.state.term;
+    var _this = this;
+    var queryString = "org:MozillaFoundation+org:mozilla+";
+    queryString += query.map(function(item) {
+      return item.searchTerm;
+    }).join("+");
+    var link = "https://github.com/issues?utf8=✓&q=" + queryString;
+    var doneButton;
+    if (query.length) {
+      doneButton = (<div className="clear-fix"><a href={link} className="pull-right button">Take me there</a></div>);
+    }
+    return (
+      <div>
+        <div className="header">
+          <h2>Issues</h2>
+        </div>
+        <div className="main issues-page">
+          <form onSubmit={this.onSubmit}>
+            <select onChange={this.onSearchChange} className="search-types">
+              <option value="assignee" selected>Assignee</option>
+              <option value="author">Author</option>
+              <option value="label">Label</option>
+              <option value="milestone">Milestone</option>
+              <option value="repo">Repo</option>
+            </select>
+            <span className="input-container">
+              <input onChange={this.onChange} className="do-search-input" placeholder="search term" value={term} list={this.state.search}/>
+            </span>
+            <datalist id="assignee">
```

Comment:
Been meaning to make a AutoCompleteDataList React thingy instead, so I guess I'll do that now seeing how you're calling me on it.